packages/jest-config/src/utils.ts

```diff
@@ -69,37 +69,51 @@ export const replaceRootDirInPath = (
   );
 };
 
-// TODO: Type as returning same type as input
-const _replaceRootDirInObject = (
+const _replaceRootDirInObject = <T extends ReplaceRootDirConfigObj>(
   rootDir: Config.Path,
-  config: any,
-): {[key: string]: unknown} => {
-  if (config !== null) {
-    const newConfig: {[key: string]: unknown} = {};
-    for (const configKey in config) {
-      newConfig[configKey] =
-        configKey === 'rootDir'
-          ? config[configKey]
-          : _replaceRootDirTags(rootDir, config[configKey]);
-    }
-    return newConfig;
+  config: T,
+): T => {
+  const newConfig = {} as T;
+  for (const configKey in config) {
+    newConfig[configKey] =
+      configKey === 'rootDir'
+        ? config[configKey]
+        : _replaceRootDirTags(rootDir, config[configKey]);
   }
-  return config;
+  return newConfig;
 };
 
-// TODO: Type as returning same type as input
-export const _replaceRootDirTags = (rootDir: Config.Path, config: any): any => {
+type ReplaceRootDirConfigObj = {[key: string]: Config.Path};
+type ReplaceRootDirConfigValues =
+  | ReplaceRootDirConfigObj
+  | RegExp
+  | Config.Path
+  | Array<ReplaceRootDirConfigObj>
+  | Array<string>;
```

Comment:
Can this be `Array<Config.Path>` as well for consistency? Also, are `Array<RegExp>`s not allowed?

Reply:
Was a little confused was Regex was even in there since nothing would be a Regex instance right?, only regex pattern strings and glob pattern strings, And those that can be array of regex patterns strings get handled in `normalizeUnmockedModulePathPatterns`. (coveragePathIgnorePatterns ,modulePathIgnorePatterns).

I also dont know why there is `_replaceRootDirInObject`, the only option would be the `moduleNameMapper`, but that option gets transformed before passing to `_replaceRootDirTags` anyway.

Current options that use  `_replaceRootDirTags` : `projects` (only passes if its a sting), `moduleDirectories`,`testMatch`,`moduleNameMapper`

Reply:
@jeysal  , i just went ahead and added it anyway.